frontend/libs/studio-components-legacy/src/components/StudioCodelistEditor/StudioCodeListEditor.test.tsx

```diff
@@ -344,6 +344,74 @@ describe('StudioCodeListEditor', () => {
     });
   });
 
+  describe('onCreateTextResource', () => {
+    it('Calls the onCreateTextResource callback with the new text resource when a help text is changed and there is no text resource linked to it', async () => {
+      const user = userEvent.setup();
+      const onCreateTextResource = jest.fn();
+      renderCodeListEditor({
+        ...propsWithTextResources,
+        codeList: codeListWithoutTextResources,
+        onCreateTextResource,
+      });
+      const testRowNumber = 1;
+      const propertyCoords: TextPropertyCoords = [testRowNumber, CodeListItemTextProperty.HelpText];
+      const newValue = 'new text';
+
+      await user.type(getTextResourceValueInput(propertyCoords), newValue);
+      await user.tab();
+
+      expect(onCreateTextResource).toHaveBeenCalledTimes(1);
+      expect(onCreateTextResource).toHaveBeenCalledWith({
+        value: expect.stringContaining(newValue),
+        id: expect.stringMatching(/^id_[1-9][0-9]{3}$/),
```

Comment:
Vi kan godt overlate detaljene her til den funksjonsspesifikke testen.
```suggestion
        id: expect.any(String),
```